cassandra_test.go

```diff
@@ -159,10 +159,12 @@ func TestObserve(t *testing.T) {
 		observedStmt = ""
 	}
 
-	observer := funcQueryObserver(func(ctx context.Context, o ObservedQuery) {
-		observedKeyspace = o.Keyspace
-		observedStmt = o.Statement
-		observedErr = o.Err
+	observer := funcQueryObserver(func(ctx context.Context, o *ObservedQuery) {
+		if o != nil {
```

Comment:
We don't need the nil check here. QueryObserver won't ever be called with nil pointer.

Reply:
Fixed and similar changes made for BatchObserver as well.

Benchmarks:
```
QueryObserver

Benchmark_GOCQL-16              613           2045540 ns/op           11650 B/op         88 allocs/op
Benchmark_ForkedGOCQL-16        858           1432294 ns/op            6051 B/op         87 allocs/op
```

```
BatchObserver

Benchmark_BatchGOCQL-16             862           1320205 ns/op            7902 B/op         31 allocs/op
Benchmark_ForkedBatchGOCQL-16       956           1297137 ns/op            2551 B/op         30 allocs/op
```